services/api-server/src/simcore_service_api_server/api/routes/files.py

```diff
@@ -244,25 +287,56 @@ async def get_file(
 ):
     """Gets metadata for a given file resource"""
 
+    return await _get_file(
+        file_id=file_id,
+        storage_client=storage_client,
+        user_id=user_id,
+        access_right="read",
+    )
+
+
+@router.get(
+    ":search",
+    response_model=Page[File],
+    responses={**_COMMON_ERROR_RESPONSES},
+)
+async def search_files_page(
+    storage_client: Annotated[StorageApi, Depends(get_api_client(StorageApi))],
+    user_id: Annotated[int, Depends(get_current_user_id)],
+    page_params: Annotated[PaginationParams, Depends()],
+    sha256_checksum: SHA256Str | None = None,
+    file_id: UUID | None = None,
+):
+    """Search files"""
+    # TODO implement the pagination of this method directly in storage(https://github.com/ITISFoundation/osparc-simcore/issues/4773)
```

Comment:
But where should I put it then? Ideally I would go ahead and actually fix the todo, but that is too far-reaching at this point. I added an issue, but I would like to keep the todo so that if we encounter issues with this in the future we see that this is likely to be the issue.

Reply:
I would only leave a NOTE (not even a TODO) if in my next PR I need to change something.
In this case, I think the issue is more than enough. At a later date we need to paginate storage and this search for sure will change then.
I think you can safely drop this comment. If we don't change it someone will close the issue and this line will stay here for a very long time or forever. (we are full of such todos, which after a while mean nothing).